databricks/koalas/indexes.py

```diff
@@ -958,6 +958,90 @@ def max(self):
 
         return result if len(result) > 1 else result[0]
 
+    def delete(self, loc):
+        """
+        Make new Index with passed location(-s) deleted.
+
+        Returns
+        -------
+        new_index : Index
+
+        Examples
+        --------
+        >>> kidx = ks.Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10])
+        >>> kidx
+        Int64Index([10, 10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete(0)
+        Int64Index([10, 9, 8, 4, 2, 4, 4, 2, 2, 10, 10], dtype='int64')
+
+        >>> kidx.delete([0, 1, 2, 3, 10, 11])
+        Int64Index([4, 2, 4, 4, 2, 2], dtype='int64')
+        """
+        if isinstance(self, MultiIndex):
+            raise NotImplementedError("currently doesn't support for MultiIndex")
+        if not is_list_like(loc):
+            if not isinstance(self, str) and abs(loc) >= len(self):
+                raise IndexError(
+                    "index {} is out of bounds for axis 0 with size {}"
+                    .format(loc, len(self)))
+            loc = [loc]
+        loc = [int(item) for item in loc]
+        loc = [item if item >= 0 else len(self) + item for item in loc]
+
+        sdf = self._internal.sdf
+        sdf = sdf.select(self.to_series()._scol.alias('__index_value__'))
+        # sdf here looks like below
+        # +---------------+
+        # |__index_value__|
+        # +---------------+
+        # |             10|
+        # |             10|
+        # |              9|
+        # |              8|
+        # |              4|
+        # |              2|
+        # |              4|
+        # |              4|
+        # |              2|
+        # |              2|
+        # |             10|
+        # |             10|
+        # +---------------+
+
+        sdf = _InternalFrame.attach_default_index(
+            sdf, default_index_type='distributed-sequence')
```

Comment:
@itholic, can you add `note::` in the docstring that says this API is based on a global sequence which is pretty expensive?

Also, don't forget that this group-apply causes OOO if one partition does not fit into executor's memory. We're using this way because there's no choice.